inaugurator/ceremony.py

```diff
@@ -313,20 +314,43 @@ def _additionalDownload(self, destination):
             downloadInstance.download(destination)
 
     def _makeSureDiskIsMountable(self):
-        if self._args.inauguratorTargetDeviceCandidate is None:
+        self._setTargetDevice()
+        self._createPartitionTable()
+        logging.info("Partitions created")
+        self._mountOp = mount.Mount()
+        assert self._bootPartitionPath is not None, "Please initialize boot partition path first"
+        self._mountOp.setBootPartitionPath(self._bootPartitionPath)
+
+    def _setTargetDevice(self):
+        if self._args.inauguratorTargetDeviceCandidate is not None:
+            logging.info("Explicit target devices indicated: %(deviceCandidates)s",
+                         dict(deviceCandidates=self._args.inauguratorTargetDeviceCandidate))
+            candidates = self._args.inauguratorTargetDeviceCandidate
+        elif self._args.inauguratorTargetDeviceType is not None:
             logging.info("Searching for target devices of type %(deviceType)s",
                          dict(deviceType=self._args.inauguratorTargetDeviceType))
             device = self._storageDevices.findFirstDeviceOfType(self._args.inauguratorTargetDeviceType,
                                                                 self._talkToServer)
+            if not device:
+                raise Exception("No %s devices found" % self._args.inauguratorTargetDeviceType)
             candidates = [device]
-        else:
-            candidates = self._args.inauguratorTargetDeviceCandidate
+        elif self._args.inauguratorTargetDeviceLabel is not None:
+            logging.info("Searching for target devices with label %(deviceLabel)s",
+                         dict(deviceLabel=self._args.inauguratorTargetDeviceLabel))
+            candidates = self._getDevicesWithLabel(self._args.inauguratorTargetDeviceLabel)
```

Comment:
After talking with Rom we've realized that this validation (no more than 1 BOOT partition) should not only happen under this option (inauguratorTargetDeviceLabel), but at any option. So, consider either:
* doing this validation with 1 tab less (validate that there are no BOOT labels on devices which are not the target device).
* Adding partitiontable.py - when verifying the partition table after creation, then if it's set to NOT wipe other boot partitions, then also raise an exception there.
* Both of the above.